File: PetriNetPlugin/src/org/workcraft/plugins/petri/tools/TracePair.java
package org.workcraft.plugins.petri.tools;

import org.workcraft.Trace;

public class TracePair {
	public final Trace trace;
	public final Trace branchTrace;
	public int traceStep;
	public int branchStep;

	public TracePair(Trace trace, int traceStep, Trace branchTrace, int branchStep) {
		this.trace = trace;
		this.branchTrace = branchTrace;
		this.traceStep = traceStep;
		this.branchStep = branchStep;
	}
	
	public static TracePair createEmpty(){
		return new TracePair(new Trace(), 0, new Trace(), 0);
	}

	public TracePair clone() {
		return new TracePair(trace.clone(), traceStep, branchTrace.clone(), branchStep);
	}
}

File: ScalaGraphEditorUtil/src/main/scala/org/workcraft/plugins/petri/tools/TracePair.scala
package org.workcraft.plugins.petri.tools

object TracePair {
  import scalaz.Lens
  def createEmpty = TracePair(TraceStep(Trace(Nil),0),TraceStep(Trace(Nil),0))
  val main : Lens[TracePair, TraceStep] = Lens(p => p.main, (p, m) => p.copy(main = m))
  val branch : Lens[TracePair, TraceStep] = Lens(p => p.branch, (p, m) => p.copy(branch = m))
}

case class TracePair(val main : TraceStep, val branch : TraceStep)

